### PR TITLE
Gevos/tracker filter button

### PR DESCRIPTION
**Pull request by gevos**

### CODE REVIEW

File: src/components/FilterButton/FilterButton.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+class FilterButton extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      filterActivated: false,
+    };
+  }
+
+  toggleFilter(routeIdentifier, trackerLayer) {
```
Comment (mhaertwig) [inline src/components/FilterButton/FilterButton.js]: `trackerLayer` is a prop. For readability let's not pass around props as function parameters.

**Comment by edellucien** [comment]
Cancel trajectoryById is implemented.
Follow bus only when more than 1px || 0.5px is not very nice unfortunately after tests. Seems that we are always late and running after the train.

File: src/components/FilterButton/FilterButton.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import React, { PureComponent } from 'react';
+import qs from 'query-string';
+import PropTypes from 'prop-types';
+import Button from 'react-spatial/components/Button';
+import filterActive from '../../images/FilterButton/filterActive.png';
+import filterInactive from '../../images/FilterButton/filterInactive.png';
```
Comment (oterral) [inline src/components/FilterButton/FilterButton.js]: Can't we have SVG , png looks ugly on high res device ?

File: src/components/FilterButton/FilterButton.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    const activated = !filterActivated;
+
+    const filterFc = TrajservLayer.createFilter(
+      undefined,
+      routeIdentifier.split('.')[0],
+    );
+    if (trackerLayer && trackerLayer.tracker) {
+      if (activated) {
+        this.updatePermalink(false);
+        trackerLayer.tracker.setFilter(filterFc);
```
Comment (oterral) [inline src/components/FilterButton/FilterButton.js]: Add a setFilter method to trackerLayer. The trackerLayer should be responsible to set properties on the tracker.

File: src/components/FollowButton/FollowButton.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        .filter(r => r.routeIdentifier === routeIdentifier);
+      const firstCoord = trajectory && trajectory.coordinate;
+      if (firstCoord) {
+        setCenter(firstCoord);
+        this.updateInterval = window.setInterval(() => {
+          const [clickedRoute] = trackerLayer.tracker
+            .getTrajectories()
+            .filter(r => r.routeIdentifier === routeIdentifier);
+          if (clickedRoute && clickedRoute.coordinate) {
+            setCenter(clickedRoute.coordinate);
```
Comment (oterral) [inline src/components/FollowButton/FollowButton.js]: there is a duplication of code here. May be a `trackerLayer.getVehicle(filterFc)`  function makes sense here. It's probably something reusable.

File: src/components/FollowButton/FollowButton.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+          }
```
Comment (oterral) [inline src/components/FollowButton/FollowButton.js]: there is a duplication of code here

**Comment by edellucien** [comment]
MR conflict resolved with master.
Style based on css and new icon updated.

**Pull request merged**